src/pkg/packager/compose.go

```diff
@@ -5,416 +5,60 @@
-	// Make sure the component we're trying to import can't be accessed.
-	if err := validate.ImportPackage(&parentComponent); err != nil {
-		return child, fmt.Errorf("invalid import definition in the %s component: %w", parentComponent.Name, err)
-	}
-
-	// Keep track of the composed components import path to build nested composed components.
-	pathAncestry := ""
-
-	// Get the component that we are trying to import.
-	// NOTE: This function is recursive and will continue getting the children until there are no more 'imported' components left.
-	child, err = p.getChildComponent(parentComponent, pathAncestry)
-	if err != nil {
-		return child, fmt.Errorf("unable to get child component: %w", err)
-	}
-
-	// Merge the overrides from the child that we just received with the parent we were provided.
-	p.mergeComponentOverrides(&child, parentComponent)
-
-	return
-}
-
-func (p *Packager) getChildComponent(parent types.ZarfComponent, pathAncestry string) (child types.ZarfComponent, err error) {
-	// Figure out which component we are actually importing.
-	// NOTE: Default to the component name if a custom one was not provided.
-	childComponentName := parent.Import.ComponentName
-	if childComponentName == "" {
-		childComponentName = parent.Name
-	}
-
-	var cachePath string
-
-	subPkgPaths := layout.New(parent.Import.Path)
-
-	if parent.Import.URL != "" {
-		if !strings.HasSuffix(parent.Import.URL, oci.SkeletonSuffix) {
-			return child, fmt.Errorf("import URL must be a 'skeleton' package: %s", parent.Import.URL)
-		}
-
-		// Save all the OCI imported components into our build data
-		p.cfg.Pkg.Build.OCIImportedComponents[parent.Import.URL] = childComponentName
-
-		skelURL := strings.TrimPrefix(parent.Import.URL, helpers.OCIURLPrefix)
-		cachePath = filepath.Join(config.GetAbsCachePath(), "oci", skelURL)
-		err = os.MkdirAll(cachePath, 0755)
-		if err != nil {
-			return child, fmt.Errorf("unable to create cache path %s: %w", cachePath, err)
-		}
+	pkgVars := p.cfg.Pkg.Variables
+	pkgConsts := p.cfg.Pkg.Constants
 
-		err = p.setOCIRemote(parent.Import.URL)
-		if err != nil {
-			return child, err
-		}
-		manifest, err := p.remote.FetchRoot()
-		if err != nil {
-			return child, err
-		}
-		tb := filepath.Join(layout.ComponentsDir, fmt.Sprintf("%s.tar", childComponentName))
-		fetchedLayers, err := p.remote.PullPackage(cachePath, config.CommonOptions.OCIConcurrency, manifest.Locate(tb))
-		if err != nil {
-			return child, fmt.Errorf("unable to pull skeleton from %s: %w", skelURL, err)
-		}
-		cwd, err := os.Getwd()
-		if err != nil {
-			return child, fmt.Errorf("unable to get current working directory: %w", err)
+	for _, component := range p.cfg.Pkg.Components {
+		arch := p.arch
+		// filter by architecture if set, otherwise use system architecture
+		if component.Only.Cluster.Architecture != "" {
+			arch = component.Only.Cluster.Architecture
```

Comment:
This check should just be filtered out - if we have a component that doesn't match the arch we are building for there is no reason to compose it.